code/script.js

```diff
@@ -1,325 +1,361 @@
+      )
+      charactersInPlay = charactersInPlay.filter((pokemon) => !pokemon[category].includes(value))
     }
-  } else if (category === 'other') {
-    // Similar to the one above
   } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+    if (keepParam) {
+      alert(
+        `Yes, the pokemon has a ${value}! Keep all pokemon that have a ${value}.`
+      )
+      charactersInPlay = charactersInPlay.filter((pokemon) => pokemon[category].includes(value))
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the pokemon doesn't have a ${value}! Remove all pokemon that have a ${value}.`
+      )
+      charactersInPlay = charactersInPlay.filter((pokemon) => !pokemon[category].includes(value))
     }
   }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+  // Invoke a function to redraw the board with the remaining pokemon.
+  generateBoard();
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+const guess = (pokemonToConfirm) => {
+  let result = confirm(`Are you sure you want to pick ${pokemonToConfirm}?`);
+  // If the player wants to guess, the checkMyGuess function is invoked.
+  if (result) {
+    checkMyGuess(pokemonToConfirm);
+  }
 }
 
-// If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+const checkMyGuess = (pokemonToCheck) => {
+  board.style.display = "none";
+  winOrLose.style.display = "flex";
+
+  winOrLoseText.innerHTML =
+    (pokemonToCheck === secretCharacter.name)
+      ? `Yay! ${pokemonToCheck} was correct! Congratz!`
+      : `Oh, no! ${secretCharacter.name} was the right answer! Better luck next time!`;
 }
 
 // Invokes the start function when website is loaded
 start()
 
+// Counter for number of guesses
+let counter = 0;
```

Comment:
I didnt understand this function, when I guessed on the Pokemons it only counts til the number 1?